mayastor/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -853,10 +854,42 @@ impl Nexus {
             .ch
             .iter()
             .map(|c| unsafe {
-                let (b, c) = c.io_tuple();
+                let (bdev, chan) = c.io_tuple();
                 spdk_bdev_unmap_blocks(
-                    b,
-                    c,
+                    bdev,
+                    chan,
+                    io.offset() + io.nexus_as_ref().data_ent_offset,
+                    io.num_blocks(),
+                    Some(Self::io_completion),
+                    pio as *mut _,
+                )
+            })
+            .collect::<Vec<_>>();
+
+        if results.iter().any(|r| *r != 0) {
+            error!(
+                "{}: Failed to submit dispatched IO {:?}",
+                io.nexus_as_ref().name,
+                pio
+            );
```

Comment:
Is it reasonable to log error messages in IO paths? I can image that if there is a bad device and app (fio) issues IOs at full speed then this will flood the log file. Perhaps the code that Chris is working on (error counters) is a better place for logging problems with IOs.

Reply:
Yes that would be more appropriate.
But at the moment we are logging errors for write and read ops as well, they occur at a higher frequency than flush ops.
So I think IMHO changing the error handling to use counts instead of logging a message done in a separate change and for all such error logging.